src/wp-includes/class-wp-script-modules.php

```diff
@@ -356,24 +465,32 @@ public function print_enqueued_script_modules() {
 	 * @since 6.5.0
 	 */
 	public function print_script_module_preloads() {
-		foreach ( $this->get_dependencies( array_unique( $this->queue ), array( 'static' ) ) as $id => $script_module ) {
+		$dependency_ids = $this->get_sorted_dependencies( array_unique( $this->queue ), array( 'static' ) );
```

Comment:
get_sorted_dependencies includes an array_unique call.

```suggestion
		$dependency_ids = $this->get_sorted_dependencies( $this->queue, array( 'static' ) );
```